doc/neps/roadmap.rst

```diff
@@ -46,17 +40,25 @@ Improvements to NumPy's performance are important to many users. We have
 focused this effort on Universal SIMD (see :ref:`NEP38`) intrinsics which
 provide nice improvements across various hardware platforms via an abstraction
 layer.  The infrastructure is in place, and we welcome follow-on PRs to add
-SIMD support across all relevant NumPy functions.
+SIMD support across relevant NumPy functionality.
+
+Transitioning from C to C++, both in the SIMD infrastructure and in NumPy
```

Comment:
Are the main drivers for this choice written somewhere (might help discussion in other packages).

Reply:
https://numpy.org/neps/nep-0054-simd-cpp-highway.html#motivation-and-scope contains some reasons. Some are specific to the SIMD framework, but some are more general. The `.c.src` templating that numpy source code has used for a long time is truly awful, and if all we did was translate that to C++ which was "C plus standard C++ templating", that alone would be a significant improvement.

I'd say that this is true pretty much anywhere one wants to support multiple data types in C.